### PR TITLE
Implement remove file room of removed file

### LINKED ISSUE #1328: Add background job to remove file rooms of removed files
## Steps to reproduce
1. Upload a file
2. Start to chat about it
3. Delete the file
4. Also delete the file from trashbin if enabled

### Expected behaviour
The room should be deleted after some time

### Actual behaviour
The room exists until the end of the instance

### PR DESCRIPTION
**Pull request by vitormattos**
## Explaning changes
* Refactored the code to reduce the quantity of query's
* Fixed a bug when get `federationManager` property and property don't exists

## To do
* [x] resolve #1328
* [x] Resolve performance problem.
  The code do a lot of queries on database. Is good reduce the numbers of queries maintaining with a lower memory usage.
* [x] unit tests

### CODE REVIEW

File: lib/BackgroundJob/RemoveEmptyRooms.php [hunk 1]
```diff
@@ -47,15 +47,23 @@ class RemoveEmptyRooms extends TimedJob {
...
+	protected $userMountCache;
 
 	protected $numDeletedRooms = 0;
 
+	/**
+	 * The room to proccess
+	 *
+	 * @var Room|null
+	 */
+	protected $room = null;
```
Comment (nickvergessen) [inline lib/BackgroundJob/RemoveEmptyRooms.php:60]: Made some bad experiences lately as cron jobs keep the state across multiple executions within the same `cron.php` call. It's not a problem here, but I would avoid the pattern. Can we instead pass around the room object?

File: lib/BackgroundJob/RemoveEmptyRooms.php [hunk 2]
```diff
@@ -77,13 +86,47 @@ protected function run($argument): void {
...
 	public function callback(Room $room): void {
-		if ($room->getType() === Room::CHANGELOG_CONVERSATION) {
+		$this->defineRoomToProcces($room);
+		$this->deleteIfIsEmpty();
+		$this->deleteIfFileIsRemoved();
+	}
+
+	private function defineRoomToProcces(Room $room) {
+		if ($room->getType() === Room::TYPE_CHANGELOG) {
+			return;
```
Comment (nickvergessen) [inline lib/BackgroundJob/RemoveEmptyRooms.php:96]: if this is true, `deleteIfIsEmpty` and `deleteIfFileIsRemoved` will again be called with the old room object

File: lib/BackgroundJob/RemoveEmptyRooms.php (hunk 1 above)
Comment (vitormattos) [reply lib/BackgroundJob/RemoveEmptyRooms.php:60]: The room object is changed on all executions. We can pass the object between the methods but will make the code biggest.

File: lib/BackgroundJob/RemoveEmptyRooms.php (hunk 2 above)
Comment (vitormattos) [reply lib/BackgroundJob/RemoveEmptyRooms.php:96]: You found a :bug: 
I will write a test to cover this scenario.

File: lib/BackgroundJob/RemoveEmptyRooms.php (hunk 1 above)
Comment (vitormattos) [reply lib/BackgroundJob/RemoveEmptyRooms.php:60]: Do you want that I thing on another approach?

Comment (nickvergessen) [reply lib/BackgroundJob/RemoveEmptyRooms.php:60]: You could just pass in $room into the other methods?

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**